GraphicsToolsProject/Plugins/GraphicsTools/Source/GraphicsTools/Public/GTProximityLightComponent.h

```diff
@@ -85,6 +85,54 @@ class GRAPHICSTOOLS_API UGTProximityLightComponent : public UGTLightComponent
 	UFUNCTION(BlueprintSetter, Category = "Light")
 	void SetOuterColor(FColor Color);
 
+	/** Gets the material parameter name array used to represent each location of a ProximityLight. */
+	UFUNCTION(BlueprintPure, Category = "Light")
+	const TArray<FName>& GetLocationParameterNames() const { return LocationParameterNames; }
+
+	/** Sets the material parameter name array used to represent each location of a ProximityLight. */
+	UFUNCTION(BlueprintSetter, Category = "Light")
+	void SetLocationParameterNames(const TArray<FName>& Names);
```

Comment:
>SetLocationParameterNames [](start = 6, length = 25)

I find it odd that each component instance keeps a writable duplicate of the list of parameter names, I see potential for misalignment. Do we need the flexibility to change parameter names or could we settle for a fixed naming convention?

Reply:
Thank you for pointing this out! The "old way" was to just have a fixed naming convention, which was simple and marvelous. But, to support [flexible proximity light counts](https://github.com/Cameron-Micka/MixedReality-GraphicsTools-Unreal/blob/public/0.2.x/Docs/ProximityLights.md#advanced-usage) or [flexible clipping primitive setups](https://github.com/Cameron-Micka/MixedReality-GraphicsTools-Unreal/blob/public/0.2.x/Docs/ClippingPrimitives.md#advanced-usage) I had to add a way specify unique parameter names per component.

Fortunately most of this is hidden to users in the "advanced properties" dropdown to avoid misalignment. But, I'm open to other suggestions for making this more robust?

Reply:
Understood. Would it make sense to hide the parameter names when not using a parameter collection override? The parameter names for the default parameter collection are hardcoded anyways, so it seems to me that allowing editing in those cases can lead to errors.

Reply:
Oh, love that idea! I'll do that.